Searching Algorithms in C

sumuhere opened the pull request:
Added some popular searching algorithms in C . 
It would be really appriciated if this PR merged.
issue: #114

pydevsg commented:
Hi, @sumuhere thanks for making this PR.
Please squash all the commits into one and add these files inside [C](https://github.com/JGEC-Winter-of-Code/JWoC_Algorithms/tree/master/C) folder.

sumuhere commented:
Hello @pydevsg I've squashed all the commits into one `Searching Algorithms in C` commit and also moved the searching algos to C/.
Let me know if anything i can do.

pydevsg commented:
Hey, @sumuhere great work.🎉 
Thanks :)
